corehq/apps/app_manager/views.py

```diff
@@ -2083,11 +2117,53 @@ def should_edit(attribute):
 
 @no_conflict_require_POST
 @require_can_edit_apps
+def edit_schedule_phases(request, domain, app_id, module_id):
+    NEW_PHASE_ID = -1
+    app = get_app(domain, app_id)
+    module = app.get_module(module_id)
+    phases = json.loads(request.POST.get('phases'))
+    changed_anchors = [(phase['id'], phase['anchor'])
+                       for phase in phases if phase['id'] != NEW_PHASE_ID]
+    all_anchors = [phase['anchor'] for phase in phases]
+    enabled = json.loads(request.POST.get('has_schedule'))
+    try:
+        module.update_schedule_phase_anchors(changed_anchors)
+        module.update_schedule_phases(all_anchors)
+        module.has_schedule = enabled
+    except ScheduleError as e:
+        return HttpResponseBadRequest(unicode(e))
```

Comment:
It could... although it seems as though all of the other errors in app_manager are handled this way IIRC (with a javascript alert), which I hate, although it does reduce the number of page loads, and maintains consistency.